cli/src/semdep/parsers/util.py

```diff
@@ -0,0 +1,214 @@
+from base64 import b16encode
+from base64 import b64decode
+from dataclasses import dataclass
+from pathlib import Path
+from re import escape
+from typing import Callable
+from typing import cast
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import TypeVar
+from typing import Union
+
+from parsy import alt
+from parsy import fail
+from parsy import line_info
+from parsy import ParseError
+from parsy import Parser
+from parsy import regex
+from parsy import string
+from parsy import success
+
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Direct
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Transitive
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Transitivity
+from semgrep.semgrep_interfaces.semgrep_output_v1 import Unknown
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__name__)
+
+
+A = TypeVar("A")
+B = TypeVar("B")
+
+Pos = Tuple[int, int]
+
+
+def not_any(chars: List[str]) -> "Parser[str]":
+    return regex(f"[^{escape(''.join(chars))}]+")
+
+
+def extract_npm_lockfile_hash(s: Optional[str]) -> Dict[str, List[str]]:
+    """
+    Go from:
+        sha512-aePbxDmcYW++PaqBsJ+HYUFwCdv4LVvdnhBy78E57PIor8/OVvhMrADFFEDh8DHDFRv/O9i3lPhsENjO7QX0+A==
+    To:
+        sha512,
```

Comment:
is something missing in this docstring after the comma?

Reply:
That's been missing since this was written ages ago lmao. Fixed